plugin/storage/badger/spanstore/reader.go

```diff
@@ -74,11 +74,18 @@ type executionPlan struct {
 }
 
 // NewTraceReader returns a TraceReader with cache
-func NewTraceReader(db *badger.DB, c *CacheStore) *TraceReader {
-	return &TraceReader{
+func NewTraceReader(db *badger.DB, c *CacheStore, prefillCache bool) *TraceReader {
+	reader := &TraceReader{
 		store: db,
 		cache: c,
 	}
+	if prefillCache {
+		reader.loadServices()
+		reader.cache.ServiceIterator(func(service string) {
```

Comment:
why do we need this indirection? `reader.loadServices()` can return a slice of service names to iterate